### PR TITLE
Initial event content processor

### PR DESCRIPTION
**Pull request by willbarton**
This is the initial work on the event processor. It takes an event from the WordPress API, pulls out the WordPress custom fields and exposes them at the top-level of the document that will be stored in Elasticsearch. This includes a test case that ensures that all the custom fields are added as such. 

This PR also includes the addition of the [flask-eventics](https://github.com/cfpb/flask-eventics) blueprint.
 
**Additions**
 
- Event content processor
- Test case for the content processor
- Event icalendar file generation (via flask_eventics
  
**Testing**
 
To run Sheer with this PR, flask-eventics will have be installed. That can be done using

`pip install git+https://github.com/cfpb/flask-eventics`

From the root of a cfgov-refresh clone, with a virtualenv capable of running Sheer and serving the site:

1. `cd _tests/processor_tests`
2. `pip install -r requirements.txt`
3. `python test_processors.py`
 
To test icalendar file generation you'll need the following environment variables set. They can be set in the same place as the `WORDPRESS` variable.

```shell
export EVENT_ICS_URL='/events_ics/<event_slug>'
export EVENT_SOURCE='http://localhost:9200/content/events/<event_slug>/_source'
export EVENT_CALENDAR_PRODID='-//Consumer Financial Protection Bureau Events//consumerfinance.gov//'
export EVENT_FIELD_SUMMARY='event_intro'
export EVENT_FIELD_DTSTART='agenda_beginning_time'
export EVENT_FIELD_DTEND='agenda_ending_time'
export EVENT_FIELD_DTSTAMP='event_date'
export EVENT_FIELD_UID='slug'
export EVENT_FIELD_ORGANIZER='CFPB'
export EVENT_FIELD_ORGANIZER_ADDR='somewhere@cfpb.gov'
export EVENT_FIELD_LOCATION='venue_name'
export EVENT_DEFAULT_ORGANIZER='Consumer Financial Protection Bureau'
export EVENT_DEFAULT_ORGANIZER_ADDR='somewhere@cfpb.gov'
export EVENT_DEFAULT_STATUS='CONFIRMED'
```

Once that is done, you'll need to run: `sheer index --processors events --reindex` (with some events in WordPress to actually index). Then you can visit http://localhost:7000/events_ics/<event slug>

**Review**

@kurtwall, can you make sure I'm handling the custom fields appropriately? It doesn't look like there should be more than one member for any of the arrays, but I could be mistaken about that. 

I'd also like the front-ends to take a look and see if there's anything major that stands out in terms of assumptions they've already made on the front-end and assumptions we're making on the backend.

- @dpford 
- @kurtwall
- @jimmynotjim 
- @sebworks 
- @anselmbradford 

**Preview**
 
Here is what the resulting document will look like in Elasticsearch:

```json
{
    "post_event_video_transcript": "", 
    "venue_address_city": "Big City", 
    "event_desc": "This is an event's description.", 
    "post_event_youtube_url": "http://youtube.com/something", 
    "taxonomy_event_date": [
        {
            "post_count": 1, 
            "id": 625, 
            "description": "", 
            "slug": "february-1-2015", 
            "title": "February 1 2015"
        }
    ], 
    "agenda_ending_time": "2015-03-25T02:00:00-0900", 
    "event_time": "2015-03-25T01:00:00-0900", 
    "venue_name": "Title of the event location", 
    "agenda_venue_location": "vanue location in agenda", 
    "pre-event_image": "", 
    "comment_count": 0, 
    "comment_status": "closed", 
    "type": "event", 
    "agenda_speaker": "Speaker Name", 
    "taxonomy_published_date": [
        {
            "post_count": 1, 
            "id": 624, 
            "description": "", 
            "slug": "january-1-2015", 
            "title": "January 1 2015"
        }
    ], 
    "reservation_contact_info": "Event contact info", 
    "live_event_preview_image": "", 
    "post_event_image": "", 
    "categories": [], 
    "post_event_speech_transcript": "", 
    "live_event_live_stream_available": "on", 
    "title_plain": "Test Event", 
    "modified": "2015-03-25 11:47:16", 
    "taxonomy_event_agenda_ending_time": [
        {
            "post_count": 1, 
            "id": 628, 
            "description": "", 
            "slug": "200am-americaadak", 
            "title": "2:00am America/Adak"
        }
    ], 
    "agenda_desc": "Agenda description", 
    "published_date": "", 
    "taxonomy_become_post_event": [
        {
            "post_count": 1, 
            "id": 630, 
            "description": "", 
            "slug": "april-1-2015-400am-americaadak", 
            "title": "April 1 2015 4:00am America/Adak"
        }
    ], 
    "post_event_flickr_url": "http://flickr.com/something", 
    "attachments": [], 
    "excerpt": "", 
    "post_event_pre-image": "", 
    "event_image": "", 
    "venue_address_address": "1234 A Boulevard", 
    "id": 1, 
    "custom_fields": {
        "post_event_flickr_url": [
            "http://flickr.com/something"
        ], 
        "become_live_event": [
            "2015-03-01T03:00:00-1000"
        ], 
        "post_event_youtube_url": [
            "http://youtube.com/something"
        ], 
        "event_information_reservation_desc": [
            "This describes how to make a reservation."
        ], 
        "event_information_venue_address_state": [
            "NY"
        ], 
        "become_live_event_timezone": [
            "a:1:{i:0;s:17:\"America/Anchorage\";}"
        ], 
        "event_information_venue_address_city": [
            "Big City"
        ], 
        "event_agenda_venue_location": [
            "vanue location in agenda"
        ], 
        "event_time": [
            "2015-03-25T01:00:00-0900"
        ], 
        "event_agenda_ending_time": [
            "2015-03-25T02:00:00-0900"
        ], 
        "event_agenda_ending_time_timezone": [
            "a:1:{i:0;s:0:\"\";}"
        ], 
        "event_agenda_speaker": [
            "Speaker Name"
        ], 
        "event_date": [
            "2015-02-01T00:00:00+0000"
        ], 
        "event_agenda_beginning_time": [
            "2015-03-25T01:00:00-0900"
        ], 
        "become_post_event": [
            "2015-04-01T04:00:00-0900"
        ], 
        "event_information_venue_name": [
            "Title of the event location"
        ], 
        "event_information_event_intro": [
            "Some event introduction."
        ], 
        "event_information_venue_address_address": [
            "1234 A Boulevard"
        ], 
        "event_agenda_beginning_time_timezone": [
            "a:1:{i:0;s:0:\"\";}"
        ], 
        "event_information_reservation_contact_info": [
            "Event contact info"
        ], 
        "live_event_live_stream_available": [
            "on"
        ], 
        "event_agenda_desc": [
            "Agenda description"
        ], 
        "become_post_event_timezone": [
            "a:1:{i:0;s:12:\"America/Adak\";}"
        ], 
        "event_information_venue_address_suite": [
            "A Suite"
        ], 
        "event_information_title": [
            "Title of the Event"
        ], 
        "event_information_event_desc": [
            "This is an event's description."
        ], 
        "published_date": [
            "2015-01-01T00:00:00+0000"
        ], 
        "event_time_timezone": [
            "a:1:{i:0;s:0:\"\";}"
        ], 
        "event_information_venue_address_zip_code": [
            "12345"
        ]
    }, 
    "event_intro": "Some event introduction.", 
    "taxonomy_event_agenda_beginning_time": [
        {
            "post_count": 1, 
            "id": 627, 
            "description": "", 
            "slug": "100am-americaadak", 
            "title": "1:00am America/Adak"
        }
    ], 
    "reservation_desc": "This describes how to make a reservation.", 
    "author": {
        "first_name": "Arthur", 
        "last_name": "Author", 
        "name": "aauthor", 
        "url": "", 
        "description": "", 
        "nickname": "aauthor", 
        "slug": "aauthor", 
        "id": 1
    }, 
    "content": "", 
    "venue_address_suite": "A Suite", 
    "dek": "", 
    "event_date": "2015-02-01T00:00:00+0000", 
    "post_event_become_post_event": "2015-04-01T04:00:00-0900", 
    "status": "publish", 
    "live_event_image": "", 
    "taxonomy_event_time": [
        {
            "post_count": 1, 
            "id": 626, 
            "description": "", 
            "slug": "100am-americaadak", 
            "title": "1:00am America/Adak"
        }
    ], 
    "parent": 0, 
    "tags": [], 
    "taxonomy_fj_tag": [], 
    "venue_address_zip_code": "12345", 
    "date": "2015-03-17T17:38:58Z", 
    "slug": "auto-draft", 
    "agenda_beginning_time": "2015-03-25T01:00:00-0900", 
    "_id": "auto-draft", 
    "url": "http://somewhere/event/test-event/", 
    "taxonomy_become_live_event": [
        {
            "post_count": 1, 
            "id": 629, 
            "description": "", 
            "slug": "march-1-2015-300am-americaadak", 
            "title": "March 1 2015 3:00am America/Adak"
        }
    ], 
    "venue_address_state": "NY", 
    "title": "Title of the Event", 
    "live_event_become_live_event": "2015-03-01T03:00:00-1000", 
    "order": 0
}
```

### CODE REVIEW

**Comment by willbarton** [comment]
This PR is ready to be reviewed again. The event mappings were broken and the processor itself wasn't being used. This is no longer the case.

File: _lib/wordpress_event_processor.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import sys
+import json
+import os.path
+import requests
+import dateutil.parser
+
+
+def posts_at_url(url):
+    """
+    Get WordPress posts as JSON from the given URL
```
Comment (anselmbradford) [inline _lib/wordpress_event_processor.py]: Minor: Could add a period at the end to be consistent with other code comments.

File: _lib/wordpress_event_processor.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    """
+    Yield documents to be indexed in Elasticsearch, one at a time.
+    """
+    for event in posts_at_url(url):
+        yield process_event(event)
+
+
+def process_event(event):
+    """
+    Process an an event as provided by the WordPress API aand return
```
Comment (anselmbradford) [inline _lib/wordpress_event_processor.py]: Minor: `Process an event`

Comment (anselmbradford) [inline _lib/wordpress_event_processor.py]: Also `aand` --> `and`

File: _lib/wordpress_event_processor.py (hunk 1 above)
Comment (anselmbradford) [inline _lib/wordpress_event_processor.py]: Ohhh... these are strings not comments, is this what's output to the console or something?

Comment (sebworks) [inline _lib/wordpress_event_processor.py]: @anselmbradford , https://docs.python.org/2/tutorial/controlflow.html#documentation-strings

Comment (anselmbradford) [inline _lib/wordpress_event_processor.py]: Thanks!

Comment (willbarton) [inline _lib/wordpress_event_processor.py]: They're Python docstrings. They create API documentation that can be viewed with `pydoc` at the command line.

Comment (anselmbradford) [inline _lib/wordpress_event_processor.py]: :muscle:

**Comment by dpford** [comment]
Very nice.  I loaded in an event and all appears to be working well.  I'm usually not _crazy_ about lambda functions (readability, etc.) but I'm really digging the conciseness.

**Comment by jimmynotjim** [comment]
Should the published date be a taxonomy? I thought we were just getting that from the default WP published date?

**Comment by jimmynotjim** [comment]
Personally, I'd group like items into parent objects, for example agenda, images, and times to make organization a bit cleaner.

**Comment by willbarton** [comment]
@jimmynotjim I can do that if it makes this easier on the frontend.

**Comment by jimmynotjim** [comment]
Typically I like to see like things grouped together to help create a visual map of how each bit of data relates to the others. I left out the taxonomy and custom field groups for brevity, but this should give a general idea. 

```
{
    "event_desc": "This is an event's description.", 
    "comment_count": 0, 
    "comment_status": "closed", 
    "type": "event", 
    "categories": [], 
    "title_plain": "Test Event", 
    "modified": "2015-03-25 11:47:16", 
    "published_date": "", 
    "attachments": [], 
    "excerpt": "", 
    "id": 1, 
    "event_intro": "Some event introduction.", 
    "author": {
        "first_name": "Arthur", 
        "last_name": "Author", 
        "name": "author", 
        "url": "", 
        "description": "", 
        "nickname": "author", 
        "slug": "author", 
        "id": 1
    }, 
    "content": "", 
    "dek": "", 
    "status": "publish", 
    "parent": 0, 
    "tags": [], 
    "taxonomy_fj_tag": [], 
    "date": "2015-03-17T17:38:58Z", 
    "slug": "auto-draft", 
    "_id": "auto-draft", 
    "url": "http://somewhere/event/test-event/", 
    "title": "Title of the Event", 
    "order": 0,
    "venue": {
        "name": "Title of the event location",
        "address": {
            "city": "Big City",
            "address": "1234 A Boulevard", 
            "suite": "A Suite", 
            "zip_code": "12345", 
            "state": "NY"
        }
    },
    "agenda": {
        "speaker": "Speaker Name", 
        "desc": "Agenda description"
    },
    "timestamps": {
        "event_doors": "2015-03-25T01:00:00-0900", 
        "agenda_start": "2015-03-25T01:00:00-0900", 
        "agenda_end": "2015-03-25T02:00:00-0900",     
        "live_start": "2015-03-01T03:00:00-1000"
        "post_event": "2015-04-01T04:00:00-0900", 
    },
    "livestream": {
        "available": true
    },
    "post_event": {
        "speech_transcript": "", 
        "youtube_url": "http://youtube.com/something", 
        "video_transcript": "", 
        "flickr_url": "http://flickr.com/something"
    },
    "images": {
        "pre_event": "",
        "event": "", 
        "live_event": "", 
        "post_event": ""
    },
    "reservation": {
        "desc": "This describes how to make a reservation.", 
        "contact_info": "Event contact info"
    }
}
```

**Comment by willbarton** [comment]
:+1: Perfect, I'll see what I can do!

**Comment by sebworks** [comment]
:+1: , Nice work @willbarton.

**Comment by jimmynotjim** [comment]
Awesome! Squash these down and then you're good to merge.

**Pull request merged**